### PR TITLE
Add staff badges to web ui

### PR DESCRIPTION
**Pull request by chandrn7**
Addresses https://github.com/tootsuite/mastodon/issues/6702. Displays appropriate staff badge on web UI profiles. Exposes staff, mod, admin status via the API.

Screen Shot 2021-05-06 at 4 41 54 PM
Screen Shot 2021-05-06 at 4 42 15   ##PM
Screen Shot 2021-05-06 at 4 53 05 PM

### CODE REVIEW

File: app/serializers/initial_state_serializer.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def meta
       mascot: instance_presenter.mascot&.file&.url,
       profile_directory: Setting.profile_directory,
       trends: Setting.trends,
+      show_staff_badge: Setting.show_staff_badge,
```
Comment (ClearlyClaire) [inline app/serializers/initial_state_serializer.rb:25]: This shouldn't be passed down to the UI but used to decide whether or not to include the badges in the REST API response.

File: app/serializers/rest/account_serializer.rb [hunk 2]
```diff
@@ -96,4 +96,16 @@ def suspended
...
+    object.user_staff?
+  end
+
+  def user_admin
+    object.user_admin?
+  end
+
+  def user_moderator
+    object.user_moderator?
+  end
```
Comment (ClearlyClaire) [inline app/serializers/rest/account_serializer.rb:110]: Those should only be returned if `Setting.show_staff_badge` is set.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Overall I'm in favor of this change, although I'm not too sure which shape the API should take, using three separate fields feels a bit much, maybe a `role` field with either role as a string would make better sense, I don't know.

In any case, the badge strings have to be translatable.

**Review by ClearlyClaire — APPROVED** [review]
Looks fine by me, @Gargron may want to comment on the API design, idk.

**Comment by chandrn7** [comment]
Any updates on this?

**Pull request closed without merging**